src/main/java/org/jabref/gui/entryeditor/LatexCitationsTabViewModel.java

```diff
@@ -0,0 +1,138 @@
+import org.jabref.model.texparser.TexParserResult;
+import org.jabref.preferences.PreferencesService;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LatexCitationsTabViewModel extends AbstractViewModel {
+
+    enum Status {
+        IN_PROGRESS,
+        CITATIONS_FOUND,
+        NO_RESULTS,
+        ERROR,
+        INACTIVE
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LatexCitationsTabViewModel.class);
+    private static final String TEX_EXT = ".tex";
+    private final BibDatabaseContext databaseContext;
+    private final PreferencesService preferencesService;
+    private final TaskExecutor taskExecutor;
+    private final StringProperty entryKey;
+    private final ObservableList<Citation> citationList;
+    private final ObjectProperty<Status> status;
+    private final StringProperty searchError;
+    private Future<?> searchTask;
+
+    public LatexCitationsTabViewModel(BibDatabaseContext databaseContext, PreferencesService preferencesService,
+                                      TaskExecutor taskExecutor) {
+        this.databaseContext = databaseContext;
+        this.preferencesService = preferencesService;
+        this.taskExecutor = taskExecutor;
+        this.entryKey = new SimpleStringProperty(null);
+        this.citationList = FXCollections.observableArrayList();
+        this.status = new SimpleObjectProperty<>(Status.INACTIVE);
+        this.searchError = new SimpleStringProperty(null);
+    }
+
+    public void init(BibEntry entry) {
+        cancelSearch();
+
+        entryKey.set(entry.getCiteKeyOptional().orElse(null));
+        if (entryKey.get() == null) {
+            searchError.set(Localization.lang("Selected entry does not have an associated BibTeX key."));
+            status.set(Status.ERROR);
+            return;
+        }
+
+        startSearch();
+    }
+
+    public ObservableList<Citation> getCitationList() {
+        return new ReadOnlyListWrapper<>(citationList);
+    }
+
+    public ObjectProperty<Status> statusProperty() {
+        return status;
+    }
+
+    public StringProperty searchErrorProperty() {
+        return searchError;
+    }
+
+    private void startSearch() {
+        searchTask = BackgroundTask.wrap(this::searchAndParse)
+                                   .onRunning(() -> status.set(Status.IN_PROGRESS))
+                                   .onSuccess(status::set)
+                                   .onFailure(error -> {
+                                       searchError.set(String.format("%s%n%n%s", error.getMessage(), error.getCause()));
+                                       status.set(Status.ERROR);
+                                   })
+                                   .executeWith(taskExecutor);
+    }
+
+    private void cancelSearch() {
+        if (searchTask == null || searchTask.isCancelled() || searchTask.isDone()) {
+            return;
+        }
+
+        searchTask.cancel(true);
```

Comment:
Sadly, canceling a task is a bit more complicated than that. The `cancel` method only sets the `isCanceled` status of the BackgroundTask to true, but does not kill the thread or otherwise stop the progress. That's something the task has to handle itself. The usual pattern is that you have a huge for loop and at the beginning of each iteration you check if the task is canceled, in which case you return the results so far.
https://github.com/JabRef/jabref/blob/8520833e684aa9e305dd8745e45696ffd6d984fb/src/main/java/org/jabref/gui/importer/UnlinkedFilesCrawler.java#L80-L90

Reply:
Interesting. I tracked many `ClosedByInterruptException` from `DefaultTexParser` when the active reference changes before ending the parser. Maybe because a new `searchTask` replaces the current `BackgroundTask searchTask`.

I also thought about including an `isCanceled()` check within the `DefaultTexParser.parse()` method. Do you agree?

Reply:
In principle, one could think about adding such a check also to the tex parser yes. But that would complicate things and I don't think the performance increase would justify the code overhead here.

Reply:
I have added a new catch in `DefaultTexParser.parse()` and, after debugging the process, I think this issue is solved now.

```
catch (ClosedChannelException e) {
    LOGGER.info("Parsing has been interrupted");
    return texParserResult;
}
```